### DISCUSSION TITLE
NX Module Federation: Use Angular as host app & Remote Applications as React or angular both

### DISCUSSION THREAD

**Discussion by KalanaPerera** [nrwl/nx#19421] [Feature Requests]
Hi Team,

First of all, Thank you for the NX <3 , 

While I was doing NX research, I found that NX Module Federation is specific to Framework, IF we start the **Host Application** as Angular, all the **remote applications** should be Angular, 

Is it possible to use both Angular and React applications under the same module federation? 

Host App: Angular
Remote APPs: Angular, React Etc

TIA

**Comment by jpretorius767** [comment]
Do you have any news on this?  I am in the same boat as I need to be able to have the host communicate with both Angular and React

**Reply by reactjsexample** [reply]
The solution is here: MFE [Angular Host with React Remote using Nx](https://medium.com/@info_35389/mfe-angular-host-with-react-remote-using-nx-61067408eeb4#06f6).

The source code is here: [mfe-angular-react-nx](https://github.com/angularexample/mfe-angular-react-nx)

**Comment by angularexample** [comment]
See this article for step-by-step instructions: [MFE Angular Host with React Remote using Nx](https://medium.com/@info_35389/mfe-angular-host-with-react-remote-using-nx-61067408eeb4#06f6)

Here is the source code for the Example App, [MFE Angular Host with React Remote](https://github.com/angularexample/mfe-angular-react-nx)

The MFE example has an Angular Host and two Angular Remotes, and a React Remote, using Nx.

Thanks to the author, JC Lango.

Please click UpVote and Like in this comment, Star the Github repo, and click Applause in the Medium article, this will help others find this solution.
